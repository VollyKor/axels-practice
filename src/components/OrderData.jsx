import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import qs from 'query-string';

import { getOrderDataFromLS } from 'helpers/localStorage';
import { POST_ORDER } from '../redux/saga';

export default function OrderData() {
    const { orderId, contactEmail, deliveryDate } = useSelector(
        (state) => state.order
    );
    const dispatch = useDispatch();

    useEffect(() => {
        const orderData = getOrderDataFromLS();
        dispatch(POST_ORDER(orderData));
    }, [dispatch]);

    const string = qs.stringify({ orderId, contactEmail, deliveryDate });

    return (
        <div className="p-4 mt-5">
            <h4 className="mb-4">Thank you for your order!</h4>
            <p className="font-weight-bold text-dark mb-2">
                Order number is: {orderId}
            </p>
            <p className="h6 mb-3">
                <span className="text-dark">{`You will recieve an email confirmation shortly to `}</span>
                <u>
                    <a type="email" href={`mailto:${contactEmail}`}>
                        {contactEmail}
                    </a>
                </u>
            </p>
            <p className="text-dark mb-0">{`Estimated delivery Day is `}</p>
            <p className="font-weight-bold text-dark">{deliveryDate}</p>

            <small>
                <u>
                    <Link
                        to={`/print?${string}`}
                        className="font-weight-bold"
                        target="blanc"
                    >
                        Print recipe
                    </Link>
                </u>
            </small>
        </div>
    );
}
