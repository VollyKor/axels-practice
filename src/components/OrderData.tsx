import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import qs from 'query-string';

import { getOrderDataFromLS } from 'helpers/localStorage';
import { POST_ORDER } from '../redux/ducks/order';

const OrderData = () => {
    const { orderId, contactEmail, deliveryDate } = useSelector(
        ({ order }: RootState) => order
    );

    const dispatch = useDispatch();

    useEffect(() => {
        const orderData: InewOrder = getOrderDataFromLS();
        dispatch(POST_ORDER(orderData));
    }, [dispatch]);

    const queryString = qs.stringify({ orderId, contactEmail, deliveryDate });

    return (
        <div className="p-4 mt-5">
            <h4 className="mb-4">Thank you for your order!</h4>
            <p className="font-weight-bold text-dark mb-2">
                Order number is: {orderId}
            </p>
            <p className="h6 mb-3">
                <span className="text-dark">{`You will recieve an email confirmation shortly to `}</span>
                <u>
                    <a type="email" href={`mailto:${contactEmail}`}>
                        {contactEmail}
                    </a>
                </u>
            </p>
            <p className="text-dark mb-0">{`Estimated delivery Day is `}</p>
            <p className="font-weight-bold text-dark">{deliveryDate}</p>

            <small>
                <u>
                    <Link
                        to={`/print?${queryString}`}
                        className="font-weight-bold"
                        target="blanc"
                    >
                        Print recipe
                    </Link>
                </u>
            </small>
        </div>
    );
};

export default OrderData;
