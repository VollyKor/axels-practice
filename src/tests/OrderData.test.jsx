import React from 'react';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import { MemoryRouter } from 'react-router-dom';

import OrderData from 'components/OrderData';

import createSagaMiddleware from 'redux-saga';

const sagaMiddleware = createSagaMiddleware();
const mockStore = configureMockStore([sagaMiddleware]);

const store = mockStore({
    order: {
        orderId: '123',
        contactEmail: '',
        deliveryDate: '',
    },
    startup: { complete: false },
});

const getWrapper = () =>
    mount(
        <Provider store={store}>
            <MemoryRouter>
                <OrderData />
            </MemoryRouter>
        </Provider>
    );

describe('OrderData testing', () => {
    it('renders correctly', () => {
        const wrapper = getWrapper();
        expect(wrapper).toMatchSnapshot();
    });

    // it('should render a startup component if startup is not complete', () => {
    //     const wrapper = getWrapper();
    //     expect(wrapper.find('p.font-weight-bold').text()).toIncludeText(
    //         '123'
    //     );
    // });
});
