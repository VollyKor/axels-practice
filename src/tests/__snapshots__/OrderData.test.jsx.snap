// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderData testing renders correctly 1`] = `
<div
  className="p-4 mt-5"
>
  <h4
    className="mb-4"
  >
    Thank you for your order!
  </h4>
  <p
    className="font-weight-bold text-dark mb-2"
  >
    Order number is: 
  </p>
  <p
    className="h6 mb-3"
  >
    <span
      className="text-dark"
    >
      You will recieve an email confirmation shortly to 
    </span>
    <u>
      <a
        href="mailto:"
        type="email"
      />
    </u>
  </p>
  <p
    className="text-dark mb-0"
  >
    Estimated delivery Day is 
  </p>
  <p
    className="font-weight-bold text-dark"
  />
  <small>
    <u>
      <a
        className="font-weight-bold"
        href="/print?contactEmail=&deliveryDate=&orderId="
        onClick={[Function]}
        target="blanc"
      >
        Print recipe
      </a>
    </u>
  </small>
</div>
`;
